subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -3671,76 +3728,86 @@ static int cmd_wifi_pmksa_flush(const struct shell *sh, size_t argc, char *argv[
 
 	return 0;
 }
-
 SHELL_STATIC_SUBCMD_SET_CREATE(
 	wifi_cmd_ap,
-	SHELL_CMD_ARG(disable, NULL, "Disable Access Point mode.\n", cmd_wifi_ap_disable, 1, 0),
+	SHELL_CMD_ARG(disable, NULL, "Disable Access Point mode.\n"
+			  "[-i, --iface=<interface index>] : Interface index.\n",
+			  cmd_wifi_ap_disable, 1, 1),
 	SHELL_CMD_ARG(enable, NULL,
-		      "-s --ssid=<SSID>\n"
-		      "-c --channel=<channel number>\n"
-		      "-p --passphrase=<PSK> (valid only for secure SSIDs)\n"
-		      "-k --key-mgmt=<Security type> (valid only for secure SSIDs)\n"
-		      "0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE-HNP, 4:SAE-H2E, 5:SAE-AUTO, 6:WAPI,"
-		      "7:EAP-TLS, 8:WEP, 9: WPA-PSK, 10: WPA-Auto-Personal, 11: DPP\n"
-		      "12: EAP-PEAP-MSCHAPv2, 13: EAP-PEAP-GTC, 14: EAP-TTLS-MSCHAPv2,\n"
-		      "15: EAP-PEAP-TLS, 20: SAE-EXT-KEY\n"
-		      "-w --ieee-80211w=<MFP> (optional: needs security type to be specified)\n"
-		      "0:Disable, 1:Optional, 2:Required\n"
-		      "-b --band=<band> (2 -2.6GHz, 5 - 5Ghz, 6 - 6GHz)\n"
-		      "-m --bssid=<BSSID>\n"
-		      "-i --ignore-broadcast-ssid=<type>. Hide SSID in AP mode.\n"
-		      "0: disabled (default)\n"
-		      "1: send empty (length=0) SSID in beacon and ignore probe request for "
-		      "broadcast SSID.\n"
-		      "2: clear SSID (ASCII 0), but keep the original length and ignore "
-		      "probe requests for broadcast SSID.\n"
-		      "[-B, --bandwidth=<bandwidth>]: 1:20MHz, 2:40MHz, 3:80MHz\n"
-		      "[-K, --key1-pwd for eap phase1 or --key2-pwd for eap phase2]:\n"
-		      "Private key passwd for enterprise mode. Default no password for private key.\n"
-		      "[-S, --wpa3-enterprise]: WPA3 enterprise mode:\n"
-		      "Default 0: Not WPA3 enterprise mode.\n"
-		      "1:Suite-b mode, 2:Suite-b-192-bit mode, 3:WPA3-enterprise-only mode.\n"
-		      "[-V, --eap-version]: 0 or 1. Default 1: eap version 1.\n"
-		      "[-I, --eap-id1...--eap-id8]: Client Identity. Default no eap identity.\n"
-		      "[-P, --eap-pwd1...--eap-pwd8]: Client Password.\n"
-		      "Default no password for eap user.\n"
-		      "-h --help (prints help)",
-		      cmd_wifi_ap_enable, 2, 45),
-	SHELL_CMD_ARG(stations, NULL, "List stations connected to the AP", cmd_wifi_ap_stations, 1,
-		      0),
+			  "-i --iface=<interface index>\n"
```

Comment:
If there is only one interface, then specifying one should be optional

Reply:
Yes, that is how it works, it's always an optional arg, I can fix if its not in few places.